src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java

```diff
@@ -314,6 +338,63 @@ public SkyValue compute(SkyKey key, Environment env)
                 getPrioritizedDetailedExitCode(causes)));
       }
 
+      Rule rule = null;
+      if (target instanceof Rule) {
+        rule = (Rule) target;
```

Comment:
Note Rule.getAssociatedRule() also works, so you can combine the conditions.

Reply:
Very nice. Thanks!